File: 2181_merge_nodes_in_between_zeros/2181_merge_nodes_in_between_zeros.dart
void main(List<String> args) {}

ListNode? mergeNodes(ListNode? head) {
  var current = head!.next;
  var ans = head;

  while (current != null) {
    if (current.val != 0) {
      head!.val += current.val;
    } else {
      if (current.next != null) {
        head = head!.next;
        head!.val = 0;
      }
    }
    current = current.next;
  }
  head!.next = null;
  return ans;
}

class ListNode {
  int val;
  ListNode? next;
  ListNode([this.val = 0, this.next]);
}

File: 2181_merge_nodes_in_between_zeros/2181_merge_nodes_in_between_zeros.go
package main

type ListNode struct {
	Val  int
	Next *ListNode
}

func mergeNodes(head *ListNode) *ListNode {
	current := head.Next
	ans := head

	for current != nil {
		if current.Val != 0 {
			head.Val += current.Val
		} else {
			if current.Next != nil {
				head = head.Next
				head.Val = 0
			}
		}
		current = current.Next
	}
	head.Next = nil
	return ans
}
